source/databricks/tests/integration/calculator/test_calculator_job.py

```diff
@@ -209,3 +214,136 @@ def test_calculator_job_input_and_output_integration_test(
+        f"{data_lake_path}/parquet_test_files/integration_events"
+    )
+
+    # Schema should match published_time_series_points_schema in time series
+    published_time_series_points_schema = StructType(
+        [
+            StructField("GsrnNumber", StringType(), True),
+            StructField("TransactionId", StringType(), True),
+            StructField("Quantity", DecimalType(18, 3), True),
+            StructField("Quality", LongType(), True),
+            StructField("Resolution", LongType(), True),
+            StructField("RegistrationDateTime", TimestampType(), True),
+            StructField("storedTime", TimestampType(), False),
+            StructField("time", TimestampType(), True),
+            StructField("year", IntegerType(), True),
+            StructField("month", IntegerType(), True),
+            StructField("day", IntegerType(), True),
+        ]
+    )
+
+    # Reads time_series_points json file into dataframe with published_time_series_points_schema and writes it to parquet
+    spark.read.schema(published_time_series_points_schema).json(
+        f"{json_test_files}/time_series_points.json"
+    ).write.mode("overwrite").parquet(
+        f"{data_lake_path}/parquet_test_files/time_series_points"
+    )
+
+
+# Code snippet from https://joelmccune.com/python-dictionary-as-object/
+class DictObj:
+    def __init__(self, in_dict: dict):
+        assert isinstance(in_dict, dict)
+        for key, val in in_dict.items():
+            if isinstance(val, (list, tuple)):
+                setattr(
+                    self, key, [DictObj(x) if isinstance(x, dict) else x for x in val]
+                )
+            else:
+                setattr(self, key, DictObj(val) if isinstance(val, dict) else val)
+
+
+@pytest.fixture
+def test_data_job_parameters(
+    test_data, databricks_path, data_lake_path, timestamp_factory
+):
+    "test_data parameter ensures that the corresponding test data has been created when using these corresponding job parameters"
+    return DictObj(
+        {
+            "data_storage_account_name": "foo",
+            "data_storage_account_key": "foo",
+            "integration_events_path": f"{data_lake_path}/parquet_test_files/integration_events",
+            "time_series_points_path": f"{data_lake_path}/parquet_test_files/time_series_points",
+            "process_results_path": f"{data_lake_path}/results",
+            "batch_id": "1",
+            "batch_grid_areas": [805, 806],
+            "batch_snapshot_datetime": timestamp_factory("2022-09-02T21:59:00.000Z"),
+            "batch_period_start_datetime": timestamp_factory(
+                "2022-04-01T22:00:00.000Z"
+            ),
+            "batch_period_end_datetime": timestamp_factory("2022-09-01T22:00:00.000Z"),
+            "time_zone": "Europe/Copenhagen",
+        }
+    )
+
+
+def test__creates_hour_csv_with_expected_columns_names(
+    spark, test_data, test_data_job_parameters, databricks_path, data_lake_path
```

Comment:
Shorten the parameters of all the test functions
```suggestion
    spark, test_data_job_parameters, data_lake_path
```